zhaquirks/thirdreality/button_v2.py

```diff
@@ -0,0 +1,92 @@
+"""Third Reality button devices."""
+
+from typing import Final
+
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import QuirkBuilder
+import zigpy.types as t
+from zigpy.zcl.clusters.general import MultistateInput
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+from zhaquirks.const import (
+    COMMAND,
+    COMMAND_DOUBLE,
+    COMMAND_HOLD,
+    COMMAND_RELEASE,
+    COMMAND_SINGLE,
+    DOUBLE_PRESS,
+    LONG_PRESS,
+    LONG_RELEASE,
+    SHORT_PRESS,
+    VALUE,
+    ZHA_SEND_EVENT,
+)
+
+MOVEMENT_TYPE = {
+    0: COMMAND_HOLD,
+    1: COMMAND_SINGLE,
+    2: COMMAND_DOUBLE,
+    255: COMMAND_RELEASE,
+}
+
+
+class MultistateInputCluster(CustomCluster, MultistateInput):
+    """Multistate input cluster."""
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        self._current_state = {}
+        super().__init__(*args, **kwargs)
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == 0x0055:
+            self._current_state[0x0055] = action = MOVEMENT_TYPE.get(value)
+            event_args = {VALUE: value}
+            if action is not None:
+                self.listener_event(ZHA_SEND_EVENT, action, event_args)
+
+            # show something in the sensor in HA
+            super()._update_attribute(0, action)
+
+
+class ThirdRealityButtonCluster(CustomCluster):
+    """Third Reality's button private cluster."""
+
+    cluster_id = 0xFF01
+
+    class AttributeDefs(BaseAttributeDefs):
+        """Define the attributes of a private cluster."""
+
+        # cancel double click
+        cancel_bouble_click: Final = ZCLAttributeDef(
```

Comment:
Typo + the comment really isn't helpful here.
```suggestion
        cancel_double_click: Final = ZCLAttributeDef(
```